### PR TITLE
javascript1-week2/andrii

### PR DESCRIPTION
**Pull request by ayaroshenkocom**
Here's my JS homework for weeл2. And I made a mistake with the branches: first I uploaded it to the branch from the week1 and only then here.

### CODE REVIEW

File: 3_javascript/javascript1/week2/script.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+let fullname1 = getFullname("Benjamin", "Hughes", true, false);
+let fullname2 = getFullname("BenjaminA", "Button", true, true);
+let fullname3 = getFullname("BenjaminA", "Button", false, false);
+let fullname4 = getFullname("", "Button", true, true);
+console.log(fullname1);
+console.log(fullname2);
+console.log(fullname3);
+console.log(fullname4);
+
```
Comment (balance3840) [inline 3_javascript/javascript1/week2/script.js:25]: Nice thought using IsFemale as a solution.

File: 3_javascript/javascript1/week2/script.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    const today = new Date().getDay(); //Get today's day
+    const eventDayIndex = (today + daysUntilEvent) % 7; // Calculate the index of the day of week of the event
+
+    const eventDay = weekdays[eventDayIndex]; // Get of the day of the event
+    return eventDay;
+}
+
+console.log("The event will take place on " + getEventWeekday(3));
+console.log("The event will take place on " + getEventWeekday(7));
+
```
Comment (balance3840) [inline 3_javascript/javascript1/week2/script.js:43]: Good job.

File: 3_javascript/javascript1/week2/script.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+// Student manager
+const class07Students = [];
+
+function addStudentToClass(studentName) {
+    if (studentName === "") {
+        console.log("Cannot add an empty string to the class");
+    } else if (class07Students.includes(studentName)) {
+        console.log(`Student ${studentName} is already in the class`);
+    } else if (class07Students.length >= 6 && studentName !== "Queen") {
```
Comment (balance3840) [inline 3_javascript/javascript1/week2/script.js:77]: Nice job here combining these two conditions.

File: 3_javascript/javascript1/week2/script.js [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+            break;
+        default:
+            console.log("Invalid candy type");
+            return; // Exit the function if the candy type is incorrect
+    }
+
+    const candyPrice = pricePerGram * weight; // Calculating the cost of sweets
+    boughtCandyPrices.push(candyPrice); // Adding cost to the array
+}
+
```
Comment (balance3840) [inline 3_javascript/javascript1/week2/script.js:129]: Here you could have used an object to define the candy price.

File: 3_javascript/javascript1/week2/script.js [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+const amountToSpend = Math.random() * 100; ////the number will represent the amount you can spend on candy
+function canBuyMoreCandy() {
+    let totalCandyPrice = 0;
+
+    // We calculate the total cost of all candies purchased
+    for (const candyPrice of boughtCandyPrices) {
```
Comment (balance3840) [inline 3_javascript/javascript1/week2/script.js:136]: Using some advanced javascript using ```const of```. Well done.

**Review by balance3840 — APPROVED** [review]

File: 3_javascript/javascript1/week2/script.js (hunk 5 above)
Comment (ayaroshenkocom) [reply 3_javascript/javascript1/week2/script.js:136]: Thank you very much.

**Pull request closed without merging**